Detections/AuditLogs/AdminPromoAfterRoleMgmtAppPermissionGrant.yaml

```diff
@@ -0,0 +1,74 @@
+  - PrivilegeEscalation
+  - Persistence
+relevantTechniques:
+  - T1098.003
+tags:
+  - SimuLand
+query: |
+  AuditLogs
+  | where LoggedByService =~ "Core Directory"
+  | where Category =~ "ApplicationManagement"
+  | where AADOperationType =~ "Assign"
+  | where ActivityDisplayName =~ "Add app role assignment to service principal"
+  | mv-expand TargetResources
+  | mv-expand TargetResources.modifiedProperties
+  | extend displayName_ = tostring(TargetResources_modifiedProperties.displayName)
+  | where displayName_ =~ "AppRole.Value"
+  | extend AppRole = tostring(parse_json(tostring(TargetResources_modifiedProperties.newValue)))
+  | where AppRole contains "RoleManagement.ReadWrite.Directory"
+  | extend InitiatingApp = tostring(parse_json(tostring(InitiatedBy.app)).displayName)
+  | extend Initiator = iif(isnotempty(InitiatingApp), InitiatingApp, tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName))
+  | extend Target = tostring(parse_json(tostring(TargetResources.modifiedProperties[4].newValue)))
+  | extend TargetId = tostring(parse_json(tostring(TargetResources.modifiedProperties[3].newValue)))
+  | summarize by bin(TimeGenerated, 1h), OperationName, Initiator, Target, TargetId, Result
+  | join (
+    AuditLogs
+    | where LoggedByService =~ "Core Directory"
+    | where Category =~ "RoleManagement"
+    | where AADOperationType in ("Assign", "AssignEligibleRole")
+    | where ActivityDisplayName has_any ("Add eligible member to role", "Add member to role")
+    | mv-expand TargetResources
+    | mv-expand TargetResources.modifiedProperties
+    | extend displayName_ = tostring(TargetResources_modifiedProperties.displayName)
+    | where displayName_ =~ "Role.DisplayName"
+    | extend RoleName = tostring(parse_json(tostring(TargetResources_modifiedProperties.newValue)))
+    | where RoleName contains "Admin"
+    | extend Initiator = tostring(parse_json(tostring(InitiatedBy.app)).displayName)
+    | extend InitiatorId = tostring(parse_json(tostring(InitiatedBy.app)).servicePrincipalId)
+    | extend TargetUser = tostring(TargetResources.userPrincipalName)
+    | extend Target = iif(isnotempty(TargetUser), TargetUser, tostring(TargetResources.displayName))
+    | extend TargetType = tostring(TargetResources.type)
+    | extend TargetId = tostring(TargetResources.id)
+    | summarize by bin(TimeGenerated, 1h), OperationName,  RoleName, Initiator, InitiatorId, Target, TargetId, TargetType, Result
+  ) on $left.TargetId == $right.InitiatorId
+  | extend TimeRoleMgGrant = TimeGenerated, TimeAdminPromo = TimeGenerated1, ServicePrincipal = Initiator1, ServicePrincipalId = InitiatorId,
+    TargetObject = Target1, TargetObjectId = TargetId1, TargetObjectType = TargetType
+  | where TimeRoleMgGrant < TimeAdminPromo
+  | project TimeRoleMgGrant, TimeAdminPromo, RoleName, ServicePrincipal, ServicePrincipalId, TargetObject, TargetObjectId, TargetObjectType
+  | extend timestamp = TimeRoleMgGrant, AccountCustomEntity = TargetObject
```

Comment:
No need for timestamp and AccountCustomEntity any longer here but not really hurting anything.   You can also just point to TargetObject in the entity mappings below.